### PR TITLE
Update kernel version to 5.2.0-m3

### LINKED ISSUE #29: Update carbon-kernel and carbon-feature-plugin
Need to update carbon-kernel version to 5.2.0-m4 and carbon-feature-plugin version to 3.0.0 with the changes introduced in the directory structure and the osgi tests.

### PR DESCRIPTION
**Pull request by lmanohara**
Resolves #29

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -370,13 +392,13 @@
...
-        <carbon.feature.plugin.version>2.0.1</carbon.feature.plugin.version>
+        <carbon.feature.plugin.version>3.0.0</carbon.feature.plugin.version>
         <commons.io.version>2.4.0.wso2v1</commons.io.version>
         <org.apache.felix.version>1.0.0</org.apache.felix.version>
         <com.zaxxer.hikaricp.version>2.4.1</com.zaxxer.hikaricp.version>
         <java.version>1.8</java.version>
-        <carbon.kernel.version>5.1.0-alpha2</carbon.kernel.version>
-        <carbon.jndi.version>1.0.0-alpha</carbon.jndi.version>
+        <carbon.kernel.version>5.2.0-m3</carbon.kernel.version>
+        <carbon.jndi.version>1.0.2-SNAPSHOT</carbon.jndi.version>
```
Comment (lmanohara) [inline pom.xml]: Please add the jndi release version after merging https://github.com/wso2/carbon-jndi/pull/33.

File: pom.xml [hunk 2]
```diff
@@ -272,6 +272,28 @@
...
+            <dependency>
+                <groupId>org.wso2.carbon</groupId>
+                <artifactId>org.wso2.carbon.pax.exam.feature</artifactId>
+                <version>${carbon.kernel.version}</version>
+                <type>zip</type>
+            </dependency>
+            <dependency>
+                <groupId>org.wso2.carbon.datasources</groupId>
+                <artifactId>wso2carbon-datasource-test</artifactId>
+                <version>${project.version}</version>
```
Comment (kasunbg) [inline pom.xml]: Instead of the ${project.version}, let's use ${carbon.datasource.version}

File: pom.xml [hunk 3]
```diff
@@ -272,6 +272,28 @@
...
+            </dependency>
+            <dependency>
+                <groupId>org.wso2.carbon</groupId>
+                <artifactId>org.wso2.carbon.container</artifactId>
+                <version>${carbon.kernel.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.wso2.carbon.datasources</groupId>
+                <artifactId>wso2carbon-datasource</artifactId>
+                <version>${project.version}</version>
```
Comment (kasunbg) [inline pom.xml]: Instead of the ${project.version}, let's use ${carbon.datasource.version}

File: pom.xml (hunk 1 above)
Comment (kasunbg) [reply pom.xml]: @ChanakaCooray adding chanaka

**Review by kasunbg — CHANGES_REQUESTED** [review]

**Comment by Niranjan-K** [comment]
@lmanohara : Migrating to Kernel 5.2.0-m3 would mean moving the datasources.xml configuration to the deployment.yaml using the Carbon Config Provider. In this PR, only the Kernel version is changed, but moving the datasource configuration is not there.

**Review by Niranjan-K — CHANGES_REQUESTED** [review]
Migration to Kernel 5.2.0-m3 would require moving the datasource configuration from XML to deployment.yaml using the carbon config provider.

File: pom.xml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
-  ~ Copyright 2016 WSO2, Inc. (http://wso2.com)
+  ~ Copyright 2017 WSO2, Inc. (http://wso2.com)
```
Comment (laki88) [inline pom.xml:3]: We do not change year for changed files.

**Comment by Niranjan-K** [comment]
Closing this PR as this commits are there in #33

**Pull request closed without merging**